src/components/layout/CyVerseAppBar.js

```diff
@@ -122,18 +127,156 @@ function CyverseAppBar(props) {
+            });
+        },
+        [analysisCustomAction]
+    );
+
+    const displayNotification = useCallback(
+        (notification, category) => {
+            let analysisStatus =
+                category.toLowerCase() ===
+                constants.notificationCategory.ANALYSIS.toLowerCase()
+                    ? notification.payload.status
+                    : "";
+
+            if (analysisStatus) {
+                displayAnalysisNotification(notification, analysisStatus);
+            } else {
+                announce({
+                    text: notification.message.text,
+                });
+            }
+        },
+        [displayAnalysisNotification]
+    );
+
+    const handleMessage = useCallback(
+        (event) => {
+            console.log(event.data);
+            let push_msg = null;
+            try {
+                push_msg = JSON.parse(event.data);
+            } catch (e) {
+                return;
+            }
+            if (push_msg.total) {
+                setUnSeenCount(push_msg.total);
+            }
+
+            const message = push_msg.message;
+            if (message) {
+                const category = message.type;
+                displayNotification(message, category);
+            }
+        },
+        [displayNotification]
+    );
+
+    React.useEffect(() => {
+        if (userProfile && userProfile.id) {
+            if (enabled) {
+                intercomLogin(
+                    userProfile.id,
+                    userProfile.attributes.email,
+                    appId,
+                    companyId,
+                    companyName
+                );
+            }
             setAvatarLetter(userProfile.id.charAt(0).toUpperCase());
+
+            let location = window.location;
+            let protocol =
+                location.protocol.toLowerCase() === "https:"
+                    ? constants.WSS_PROTOCOL
+                    : constants.WS_PROTOCOL;
+            let host = location.hostname;
+            let port = location.port;
+            const notificationUrl =
+                protocol +
+                host +
+                (port ? ":" + port : "") +
+                NavigationConstants.NOTIFICATION_WS;
+            console.log("Connecting websocket to " + notificationUrl);
+
+            const ws = new Sockette(notificationUrl, {
+                maxAttempts: 10,
+                onopen: (e) => {
+                    console.log("Websocket connected!");
+                    ws.send("Connected by " + userProfile.id);
+                },
+                onmessage: (e) => handleMessage(e),
+                onreconnect: (e) => console.log("Reconnecting...", e),
+                onmaximum: (e) => console.log("Stop Attempting!", e),
+                onclose: (e) => console.log("Closed!", e),
+                onerror: (e) => console.log("Error:", e),
+            });
+            return () => {
+                intercomLogout();
+                ws.close();
```

Comment:
Clean up. Logout from intercom and close websocket connection when component is unmounted.